infoschema/builder.go

```diff
@@ -122,8 +125,17 @@ func (b *Builder) ApplyDiff(m *meta.Meta, diff *model.SchemaDiff) ([]int64, erro
 			// Reduce the impact on DML when executing partition DDL. eg.
 			// While session 1 performs the DML operation associated with partition 1,
 			// the TRUNCATE operation of session 2 on partition 2 does not cause the operation of session 1 to fail.
-			if diff.Type == model.ActionTruncateTablePartition {
+			switch diff.Type {
+			case model.ActionTruncateTablePartition:
 				tblIDs = append(tblIDs, opt.OldTableID)
+				b.applyPlacementDelete(placement.GroupID(opt.OldTableID))
+				err := b.applyPlacementUpdate(placement.GroupID(opt.TableID))
+				if err != nil {
+					return nil, errors.Trace(err)
+				}
+				continue
+			case model.ActionDropTablePartition:
+				b.applyPlacementDelete(placement.GroupID(opt.OldTableID))
 				continue
```

Comment:
```suggestion
			case model.ActionDropTablePartition:
				b.applyPlacementDelete(placement.GroupID(opt.OldTableID))
```

Reply:
Continue is intended. Otherwise this PR changes the original behavior.